src/main/scala/io/qbeast/spark/QbeastTable.scala

```diff
@@ -65,17 +73,32 @@ class QbeastTable private (
    * @return the sequence of cubes to optimize in string representation
    */
   def analyze(revisionID: RevisionID): Seq[String] = {
-    AnalyzeTableCommand(getAvailableRevision(revisionID), indexedTable)
+    checkRevisionAvailable(revisionID)
+    AnalyzeTableCommand(revisionID, indexedTable)
       .run(sparkSession)
       .map(_.getString(0))
   }
 
   def analyze(): Seq[String] = {
-    AnalyzeTableCommand(latestRevisionAvailableID, indexedTable)
+    analyze(latestRevisionAvailableID)
+  }
+
+  /**
+   * The compact operation should compact the small files in the table
+   * @param revisionID the identifier of the revision to optimize.
+   *                        If doesn't exist or none is specified, would be the last available
+   */
+  def compact(revisionID: RevisionID): Unit = {
+    checkRevisionAvailable(revisionID)
+    CompactTableCommand(revisionID, indexedTable)
       .run(sparkSession)
       .map(_.getString(0))
   }
 
+  def compact(): Unit = {
+    compact(latestRevisionAvailableID)
```

Comment:
I like this approach, it would be nice to apply it to the analyze methods as well. I mean that operation without revision identifier calls the operation with revision passing the latest one.

Reply:
I did the same for analyze() and optimize(). Doesn't it appear on the changes?